atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2538,7 +2547,8 @@ private LongSet getStartTimestampsForValues(Iterable<Value> values) {
      * @throws TransactionLockTimeoutException If our locks timed out while trying to commit.
      * @throws TransactionCommitFailedException failed when committing in a way that isn't retriable
      */
-    private void putCommitTimestamp(long commitTimestamp, LockToken locksToken, TransactionService transactionService)
+    private void putCommitTimestamp(
+            long commitTimestamp, Set<LockToken> locksToken, TransactionService transactionService)
```

Comment:
At one point I understood WHY we're passing the TransactionService explicitly here...

Reply:
But I think this would be nice to unwind/find a better implementation

Reply:
Will send you an internal link I just found :P